tests/potential/test_potential.py

```diff
@@ -0,0 +1,294 @@
+        self.assertCountEqual(m.pairs, pairs)
+
+        #test construction with mixed tuple/list input
+        m = potential.CoefficientMatrix(types=('A','B'), params=['energy','mass'])
+        self.assertEqual(m.types, types)
+        self.assertEqual(m.params, params)
+        self.assertCountEqual(m.pairs, pairs)
+
+        #test construction with default values initialized
+        default = {'energy':0.0, 'mass':0.0}
+        m = potential.CoefficientMatrix(types=('A','B'), params=('energy','mass'),
+                                        default={'energy':0.0, 'mass':0.0})
+        self.assertEqual(m.types, types)
+        self.assertEqual(m.params, params)
+        self.assertEqual(m.default.todict(), default)
+        self.assertCountEqual(m.pairs, pairs)
+
+        #test construction with int type parameters
+        with self.assertRaises(TypeError):
+            m = potential.CoefficientMatrix(types=('A','B'), params=(1,2))
+
+        #test construction with mixed type parameters
+        with self.assertRaises(TypeError):
+            m = potential.CoefficientMatrix(types=('A','B'), params=('1',2))
+
+    def test_accessor_methods(self):
+        """Test various get and set methods on pairs"""
+        m = potential.CoefficientMatrix(types=('A',), params=('energy','mass'))
+
+        #test set and get for each pair type and param
+        self.assertEqual(m['A','A']['energy'], None)
+        self.assertEqual(m['A','A']['mass'], None)
+
+        #test setting all key values at once
+        m['A','A'] = {'energy':1.0, 'mass':2.0}
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 2.0)
+
+        #test setting key values partially with = operator
+        m['A','A'] = {'energy':1.5}
+        self.assertEqual(m['A','A']['energy'], 1.5)
+        self.assertEqual(m['A','A']['mass'], None)
+
+        m['A','A'] = {'mass':2.5}
+        self.assertEqual(m['A','A']['energy'], None)
+        self.assertEqual(m['A','A']['mass'], 2.5)
+
+        #test setting key values partially using update()
+        m['A','A'].update(energy=2.5)  #using keyword arg
+        self.assertEqual(m['A','A']['energy'], 2.5)
+        self.assertEqual(m['A','A']['mass'], 2.5)
+
+        m['A','A'].update({'mass':0.5})  #using dict
+        self.assertEqual(m['A','A']['energy'], 2.5)
+        self.assertEqual(m['A','A']['mass'], 0.5)
+
+        #test accessing key param values individually
+        m['A','A']['energy'] = 1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 0.5)
+
+        m['A','A']['mass'] = 2.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 2.0)
+
+        #test reset and get via iteration
+        params = ('energy', 'mass')
+        for j in params:
+            m['A','A'][j] = 1.5
+        self.assertEqual(m['A','A']['energy'], 1.5)
+        self.assertEqual(m['A','A']['mass'], 1.5)
+
```

Comment:
Can you add some tests here where you both get / set on an invalid (1) type pair and (2) parameter name?